features/dd-sdk-android-flags/src/main/kotlin/com/datadog/android/flags/featureflags/internal/NoOpFlagsClient.kt

```diff
@@ -15,54 +16,97 @@ import org.json.JSONObject
+    }
 
     /**
      * Returns the provided default value without any flag evaluation.
      * @param flagKey Ignored flag key.
      * @param defaultValue The value to return.
      * @return The provided default value.
      */
-    override fun resolveBooleanValue(flagKey: String, defaultValue: Boolean): Boolean = defaultValue
+    override fun resolveBooleanValue(flagKey: String, defaultValue: Boolean): Boolean {
+        logCriticalError("resolveBooleanValue for flag '$flagKey'")
+        return defaultValue
+    }
 
     /**
      * Returns the provided default value without any flag evaluation.
      * @param flagKey Ignored flag key.
      * @param defaultValue The value to return.
      * @return The provided default value.
      */
-    override fun resolveStringValue(flagKey: String, defaultValue: String): String = defaultValue
+    override fun resolveStringValue(flagKey: String, defaultValue: String): String {
+        logCriticalError("resolveStringValue for flag '$flagKey'")
+        return defaultValue
+    }
 
     /**
      * Returns the provided default value without any flag evaluation.
      * @param flagKey Ignored flag key.
      * @param defaultValue The value to return.
      * @return The provided default value.
      */
-    override fun resolveDoubleValue(flagKey: String, defaultValue: Double): Double = defaultValue
+    override fun resolveDoubleValue(flagKey: String, defaultValue: Double): Double {
+        logCriticalError("resolveDoubleValue for flag '$flagKey'")
+        return defaultValue
+    }
 
     /**
      * Returns the provided default value without any flag evaluation.
      * @param flagKey Ignored flag key.
      * @param defaultValue The value to return.
      * @return The provided default value.
      */
-    override fun resolveIntValue(flagKey: String, defaultValue: Int): Int = defaultValue
+    override fun resolveIntValue(flagKey: String, defaultValue: Int): Int {
+        logCriticalError("resolveIntValue for flag '$flagKey'")
+        return defaultValue
+    }
 
     /**
      * Returns the provided default value without any flag evaluation.
      * @param flagKey Ignored flag key.
      * @param defaultValue The value to return.
      * @return The provided default value.
      */
-    override fun resolveStructureValue(flagKey: String, defaultValue: JSONObject): JSONObject = defaultValue
+    override fun resolveStructureValue(flagKey: String, defaultValue: JSONObject): JSONObject {
+        logCriticalError("resolveStructureValue for flag '$flagKey'")
+        return defaultValue
+    }
+
+    /**
+     * Logs a critical error with both USER and MAINTAINER targets.
+     * This ensures visibility in both debug builds (MAINTAINER) and production (USER, if verbosity allows).
+     *
+     * @param operation The operation being called (e.g., "resolveBooleanValue for flag 'my-flag'")
+     */
+    private fun logCriticalError(operation: String) {
```

Comment:
is it useful to log in the `NoOpFlagsClient` given that we will already print a warning to logcat when this client is created?